cypress/e2e/Admin/results/TableByQuestion.cy.js

```diff
@@ -0,0 +1,203 @@
+
+    // Test that each table are correctly displayed
+    APP_SETTINGS2.filter((s) => s.name === APP_SETTING_NAMES.QUESTION).forEach(
+      (s, idx) => {
+        cy.get(dataCyWrapper(buildTableByQuestionCy(s.data.question))).should(
+          'have.text',
+          APP_SETTINGS2[idx].data.question
+        );
+
+        // test header
+        testTableHeader(s.data.question, 'sorted ascending');
+        // test content
+        testTableContent(s.data.question, true);
+
+        // sort descending
+        cy.get(
+          dataCyWrapper(buildTableByQuestionUserHeader(s.data.question))
+        ).click();
+
+        // test header
+        testTableHeader(s.data.question, 'sorted descending');
+        // test content
+        testTableContent(s.data.question, false);
+      }
+    );
+  });
+});
+
+/**
+ * Helper function to test that the header of the table is correct
+ *
+ * @param {string} qTitle The title of the question for which we currently are displaying data
+ * @param {string} ascending Suffix to add to User whether it is sorted ascending or descending
+ */
+const testTableHeader = (qTitle, ascending) => {
+  cy.get(dataCyWrapper(buildTableByQuestionUserHeader(qTitle))).should(
+    'have.text',
+    `User${ascending}`
+  );
+  cy.get(dataCyWrapper(buildTableByQuestionAnswerHeader(qTitle))).should(
+    'have.text',
+    'Answer'
+  );
+  cy.get(dataCyWrapper(buildTableByQuestionDateHeader(qTitle))).should(
+    'have.text',
+    'Date'
+  );
+  cy.get(dataCyWrapper(buildTableByQuestionCorrectHeader(qTitle))).should(
+    'have.text',
+    'Correct'
+  );
+};
+
+/**
+ * Helper function to test that the content of the table is correct
+ *
+ * @param {string} qTitle The title of the question for which we currently are displaying data
+ * @param {boolean} ascending Whether the current sorting order is ascending or descending
+ */
+const testTableContent = (qTitle, ascending) => {
+  const n = ascending ? 0 : 2;
+  cy.get(dataCyWrapper(buildTableByQuestionTableBodyCy(qTitle)))
+    .children('tr')
+    .each((tr, idxTr) => {
+      // test for the header of the row
+      cy.wrap(tr)
+        .children('th')
+        .first()
+        .should('have.text', RESPONSES[qTitle][Math.abs(n - idxTr)].userId);
+
+      // test for the content of the row
+      cy.wrap(tr)
+        .children('td')
+        .each((td, idxTd) => {
+          if (idxTd === 2) {
+            cy.wrap(td)
+              .children('svg')
+              .first()
+              .should(
+                'have.attr',
+                'data-testid',
+                RESPONSES[qTitle][Math.abs(n - idxTr)].fields[idxTd]
+              );
+          } else {
+            cy.wrap(td).should(
+              'have.text',
+              RESPONSES[qTitle][Math.abs(n - idxTr)].fields[idxTd]
+            );
+          }
+        });
+    });
+};
+
+/**
+ * Array containing the expected data for each table in the table by question page
+ */
+const RESPONSES = {
```

Comment:
These mock data are called `fixtures`. You can either add them in `cypress/fixtures` or create a folder in the corresponding test folder.

Reply:
I moved it under the `fixtures` folder, thanks for pointing it out.